common/placer1.cc

```diff
@@ -552,7 +552,7 @@ class SAPlacer
             add_move_cell(moveChange, other_cell, newBel);
         }
 
-        if (!ctx->isBelLocationValid(newBel) || ((other_cell != nullptr && !ctx->isBelLocationValid(oldBel)))) {
+        if (!ctx->isBelLocationValid(newBel) || !ctx->isBelLocationValid(oldBel)) {
```

Comment:
I think it is worth adding a comment why this changed.  I could see it changing back otherwise.